python_modules/dagster/dagster/_core/remote_representation/external.py

```diff
@@ -80,6 +82,30 @@
     from dagster._core.scheduler.instigation import InstigatorState
     from dagster._core.snap.execution_plan_snapshot import ExecutionStepSnap
 
+_DELIMITER = "::"
+
+
+@dagster_model
+class CompoundID:
```

Comment:
this name feels a little generic considering it's specific to instigators.

Reply:
`Repository` uses it too which is why i dropped instigator framing 
open to other names tho

Reply:
CompoundExternalId
OriginAndSelectorId

meh, not user facing / changeable in the future since not persisted